File: IT/Strutture dati/Java/TestEnum.java
/* file TestEnum.java */

import java.util.Random;

enum Color {
	RED,
	YELLOW,
	GREEN
}

public class TestEnum {
	public static void main(String [] args) {
		/* inizializza casualmente il colore del semaforo */

		Random randomizer = new Random();
		int currentColor = randomizer.nextInt(3);

		/* mappa un intero al valore enumerazione appropriato */

		Color semaphore = Color.values()[currentColor];

		/* confronta il colore del semaforo con tutti i colori possibili */

		switch (semaphore)
		{
			case RED:
				System.out.println("Semaforo ROSSO: fermati!");
				break;
			case YELLOW:
				System.out.println("Semaforo GIALLO o lampeggiante: prudenza.");
				break;
			case GREEN:
				System.out.println("Semaforo VERDE: passa senza problemi...");
				break;
			default:
				System.out.println("Semaforo guasto?");
		}
	}
}

File: IT/Strutture dati/Python/testEnum.py
# file testEnum.py

from __future__ import print_function
from enum import Enum
from random import randint

class Color(Enum):
	RED = 0
	YELLOW = 1
	GREEN = 2

# inizializza casualmente il colore del semaforo

currentColor = Color(randint(0, 2))

# confronta il colore del semaforo con tutti i colori possibili

if currentColor is Color.RED:
	print("Semaforo ROSSO: fermati!")
elif currentColor is Color.YELLOW:
	print("Semaforo GIALLO o lampeggiante: prudenza.")
elif currentColor is Color.GREEN:
	print("Semaforo VERDE: passa senza problemi...")
else:
	print("Semaforo guasto?")

